File: 프로그래머스/lv3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    const length = computers.length;
    const visited = Array.from({length : n}, () => false);

    function dfs(idx) {
        visited[idx] = true;
        
        for (let i = 0; i < length; i++) {
            if (computers[idx][i] && !visited[i]) {
                dfs(i);
            }
        }
    }
    for (let i = 0; i < length; i++) {
        if (!visited[i]) {
            dfs(i);
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    visited = [False] * n
    networks = 0

    def bfs(start):
        queue = deque([start])
        visited[start] = True
        
        while queue:
            node = queue.popleft()
            for neighbor in range(n):
                if computers[node][neighbor] == 1 and not visited[neighbor]:
                    visited[neighbor] = True
                    queue.append(neighbor)

    for i in range(n):
        if not visited[i]:
            bfs(i)
            networks += 1

    return networks